### PR TITLE
BookStore:Connect to api

### PR DESCRIPTION
**Pull request by cynthiainga**
Project requirements:

- [ ] Add redux-thunk to the project.
- [ ] Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) for making API calls.
- [ ] Fetch book list from Bookstore API on Books page load.
- [ ] Refactor the add book and remove book features to persist the changes in the server.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,29 +1,72 @@
...
+    method: 'DELETE',
+    headers: HEADERS,
+    body: JSON.stringify({ item_id: bookId }),
+  })
+    .then((response) => response.text())
+    .then(
+      () => dispatch({ type: REMOVE_BOOK, payload: bookId }),
+      () => dispatch({ type: REMOVE_BOOK, payload: null }),
+    );
+};
```
Comment (FrederickMih) [inline src/redux/books/books.js:33]: - [ ] Hi @cynthiainga, please don't forget that even after refactoring your code to use the API, the Add book and Remove book features should work as before. I can't remove a particular book. When I click on the remove book button, all the books are removed. Please let's fix this issue.

Kindly make sure that each book has a unique ID

Screenshot from 2022-03-31 11-16-50

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @cynthiainga  , 

You have done a fantastic job so far in this project. Keep up! :rocket: 
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights

:arrow_forward:  Git/GitHub workflow has been followed :heavy_check_mark: 
:arrow_forward:  Linters are correctly set up :heavy_check_mark: 
:arrow_forward:  Pull Request has a title and summary :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ Optional ] Your project still contains boilerplates from CRA in the public folder. Please make sure that you get rid of these files and images since they are not serving you for now.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (cynthiainga) [reply src/redux/books/books.js:33]: Thanks for reviewing my code, I made the change now the user can delete only one book, I fixed the error.

**Review by medaminedev66 — APPROVED** [review]
## Hi @cynthiainga, 

# Status: Approved ✔️
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚢
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cynthiainga** [comment]
Thanks @medaminedev66 :pray:

**Pull request merged**